src/golang/cmd/migrator/versions/000019_add_serialization_value_to_param_op/utils.go

```diff
@@ -0,0 +1,110 @@
+package _000019_add_serialization_value_to_param_op
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/utils"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	serilization_type_key   = "serialization_type"
+	value_key               = "val"
+	spec_field              = "spec"
+	pythonExecutorPackage   = "aqueduct_executor"
+	typeInferencePythonPath = "migrators.parameter_val_type_inference_000019.main"
+)
+
+type Operator struct {
+	Id     uuid.UUID `db:"id" json:"id"`
+	OpSpec Spec      `db:"spec" json:"spec"`
+}
+
+func getAllOperators(
+	ctx context.Context,
+	db database.Database,
+) ([]Operator, error) {
+	query := "SELECT id, spec FROM operator;"
+
+	var response []Operator
+	err := db.Query(ctx, &response, query)
+	return response, err
+}
+
+func updateParamOperatorWithNewSpec(
+	ctx context.Context,
+	operator Operator,
+	db database.Database,
+) error {
+	param_val := operator.OpSpec.Param[value_key]
+	// Launch the Python job to infer the type of the parameter value
+	cmd := exec.Command(
+		"python3",
+		"-m",
+		fmt.Sprintf("%s.%s", pythonExecutorPackage, typeInferencePythonPath),
+		"--val",
+		param_val,
+	)
+	cmd.Env = os.Environ()
+
+	var outb, errb bytes.Buffer
+	cmd.Stdout = &outb
+	cmd.Stderr = &errb
+
+	err := cmd.Run()
+	if err != nil {
+		log.Errorf("Error running Python migration job. Stdout: %s, Stderr: %s.", outb.String(), errb.String())
+		return err
+	}
+
+	param_type := strings.TrimSuffix(outb.String(), "\n")
+	operator.OpSpec.Param[serilization_type_key] = param_type
+
+	// We also change the param value to be a base64 encoding
+	operator.OpSpec.Param[serilization_type_key] = param_type
+
+	base64val := base64.StdEncoding.EncodeToString([]byte(param_val))
```

Comment:
I think `param_val` is a json-serialized string, so I'm not sure whether directly converting it to byte and then base64 encoding to a string works. Maybe we should let the conversion happen entirely in the Python script: the script first do `json.loads(param_val)`, and then do a base64 encode. And then the Golang side picks up this value and store back into the db.